File: 41和为S的连续正数序列/Solution.java
import java.util.ArrayList;
public class Solution {
    public ArrayList<ArrayList<Integer> > FindContinuousSequence(int sum) {
       ArrayList<ArrayList<Integer> > result=new ArrayList<> ();
       if (sum<=2) return result;
       int low=1,high=2;
       while (high>low){
           int num=(high+low)*(high-low+1)>>1;
           if (sum>num){
               high++;
           }
           else if(sum<num){
               low++;
           }
           else if (sum==num){
               ArrayList<Integer> temp =new ArrayList<Integer> ();
               for (int i=low;i<=high;i++) temp.add(i);
               result.add(temp);
               low++;
           }
           
       }
        return result;
    }
}

File: 41和为S的连续正数序列/Solution.cpp
class Solution {
public:
    vector<vector<int> > FindContinuousSequence(int sum) {
        vector<vector<int>> result;
        int high=2,low=1;
        while (high>low){
           int curSum=(low+high)*(high-low+1)>>1;
            if (curSum<sum){
                high++;
            }
            if (curSum==sum){
                vector<int> temp;
                for (int i=low;i<=high;i++){
                    temp.push_back(i);
                }
                result.push_back(temp);
                low++;
            }
            if (curSum>sum){
                low++;
            }
        }
        return result;
    }
};
/* 这题妥妥的双指针，然后利用等差数列的求和公式，不断去更新先后指针
的值，就可以得到所有的需要的值。 */